src/services/balancer/pools/exits/handlers/generalised-exit.handler.ts

```diff
@@ -55,40 +49,47 @@ export class DeepExitHandler implements ExitPoolHandler {
     slippageBsp,
     relayerSignature,
   }: ExitParams): Promise<QueryOutput> {
-    const bnumAmount = parseFixed(
+    const evmAmountIn = parseFixed(
       bptIn || '0',
       this.pool.value.onchain?.decimals ?? 18
     );
 
-    if (bnumAmount.lte(0)) throw new Error('BPT in amount is 0.');
+    if (evmAmountIn.lte(0)) throw new Error('BPT in amount is 0.');
 
     const signerAddress = await signer.getAddress();
-
     const slippage = slippageBsp.toString();
-    const poolId = this.pool.value.id;
-
-    this.lastGeneralisedExitRes = await balancer.pools
-      .generalisedExit(
-        poolId,
-        bnumAmount.toString(),
-        signerAddress,
-        slippage,
-        relayerSignature
-      )
-      .catch(err => {
-        console.error(err);
-        throw new Error(err);
-      });
-    if (!this.lastGeneralisedExitRes) throw new Error('Not enough liquidity.');
-
-    const tokenAddressesOut = this.lastGeneralisedExitRes.tokensOut;
-    const allPoolTokens = flatTokenTree(this.pool.value);
+
+    this.lastExitRes = await balancer.pools.generalisedExit(
+      this.pool.value.id,
+      evmAmountIn.toString(),
+      signerAddress,
+      slippage,
+      relayerSignature
+    );
+    if (!this.lastExitRes) throw new Error('Failed to query exit.');
```

Comment:
It previously said "Not enough liquidity". Was that not true? Is there a way to tell the difference between a failed query and not enough liquidity? Does the user actually see this?

Reply:
The only time 'Not enough liquidity' is relevant is for swap join/exits which are handled in the swap join/exit handlers. This error message looks like it was copied over without understanding why it was defined.

Yes, the user sees these error messages, so maybe 'Failed to query exit' is not perfect.